zadania_ruby/oop/products.rb

```diff
@@ -0,0 +1,84 @@
+class Products
+  def initialize
+    @products = []
+    load_products
+  end
+
+  def load_products
+    if @products == []
+      CSV.foreach('products.csv', headers: true) do |row|
+        @products << parse_line(row)
+      end
+    end
+    @products
+  end
+
+  def find(product)
+    result = nil
+    @products.each do |row|
+      if row[:name] == product
+        result = "#{row[:name]} #{row[:price]}#{row[:currency]}"
+      end
+    end
+    result
+  end
+
+  def find_cheaper_products(price)
+    @products.find_all do |row|
+      row[:price].to_f < price.to_f
+    end.each do |row|
+      puts "#{row[:name]} #{row[:price]}#{row[:currency]}\n"
+    end
+  end
+
+  def find_more_expensive_products(price)
+    @products.find_all do |row|
+      if row[:price].to_f > price.to_f
+        puts "#{row[:name]} #{row[:price]}#{row[:currency]}"
+      end
+    end
+  end
+
+  def convert_prices_to_currency(exchange_rate, currency, file_name)
+    new_headers = ['Product', "Price(#{currency})", 'Weight(kg)']
+    CSV.open("./#{file_name}", 'w') do |csv|
+      csv << new_headers
+      @products.each do |row|
+        new_price = row[:price].to_f / exchange_rate.to_f
+        csv << ["#{row[:name]}", "#{new_price.round(2)}", "#{row[:weight]}"]
```

Comment:
wystarczy `row[:name]` bez zamykania tego w stringu, to samo tyczy się row weight i new_rpice